omnisafe/adapter/offpolicy_adapter.py

```diff
@@ -61,15 +59,11 @@ def roll_out(  # pylint: disable=too-many-locals
             use_rand_action (bool): Whether to use random action.
         """
         if use_rand_action:
-            if isinstance(self._env.action_space, spaces.Box):
-                act_fn = partial(
-                    torch.rand, size=(self._env.num_envs, *self._env.action_space.shape)
-                )
+            act = torch.as_tensor(self._env.sample_action(), dtype=torch.float32).to(self._device)
         else:
-            act_fn = partial(agent.step, self._current_obs, deterministic=False)
+            act = agent.step(self._current_obs, deterministic=False)
 
         for _ in range(roll_out_step):
-            act = act_fn()
             next_obs, reward, cost, terminated, truncated, info = self.step(act)
```

Comment:
in all loop of `roll_out_step`, use same `act`? it seem like a bug, we must consider the situation `roll_out_step` != 1